src/hash.hpp

```diff
@@ -42,14 +44,25 @@ bool digests_equal(const struct digest* d1, const struct digest* d2);
 // struct hash represents the hash algorithm's inner state.
 struct hash;
 
-// Create a new hash state.
-struct hash* hash_init(void);
+// value semantic hash to replace 'struct hash' in the future
+class Hash {
```

Comment:
How about implementing the class in `Hash.hpp`/`Hash.cpp` instead?

Reply:
You are technically quite right, but as not everyone has a filesystem which is case sensitive it feels like this should be avoided. I would therefore rename the files as a last step.

Reply:
Right, I solved that problem for utils by renaming utils to legacy_utils. But OK.